### PR TITLE
Use publisher instance instead of global publisher object

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

**Comment by ruflin** [comment]
@urso will this have any impact from your point of view?

**Comment by urso** [comment]
Looks OK.  But why not have a publisher.New instead of allocating 'foreign' structure and calling in it on it.

**Comment by ruflin** [comment]
Should we just rename Init to New or split it up?

**Comment by urso** [comment]
New should allocate and do the init. Let's not export the Init method.

File: libbeat/outputs/outputs.go [hunk 1]
```diff
@@ -101,7 +100,11 @@ func InitOutputs(
 			continue
 		}
 
-		output, err := plugin.NewOutput(beat, &config, topologyExpire)
+		if config.Index == "" {
+			config.Index = beatName
```
Comment (urso) [inline libbeat/outputs/outputs.go]: +1 for setting the default here.

**Comment by urso** [comment]
LGTM. just waiting for travis.

**Pull request merged**